### ISSUE TITLE
Result is very very bad!

### ISSUE THREAD

**Issue by jinfagang** [rwightman/efficientdet-pytorch#96]
![image](https://user-images.githubusercontent.com/21303438/95643793-166d1180-0ae4-11eb-93e7-5e0a43b38203.png)

is the loss value normal?? The AP is very very very bad! i have using pretrained model.

**Comment by rwightman** [comment]
I doubt it's an issue with the code here, likely the usage. You could try the current branch I'm working on https://github.com/rwightman/efficientdet-pytorch/tree/more_datasets ... it has some features that should make fine-tuning a bit easier and may improve convergence a bit, but with the results in your picture there is likely something more significantly wrong with your dataset or setup.

**Comment by jinfagang** [comment]
@rwightman  Question is, same dataset, with another model in detectron2, works perfectly....

**Comment by rwightman** [comment]
Things are a little different here than in Detectron or Torchvision obj detection models, it's sort of half-way between the Tensorflow models approach and the typical PyTorch approach. The models here take yxyx coords for bbox, if that wasn't clear from my dataset/pipeline. Output is xyxy for my new branch, (or xywh for master) since I have to switch to that for builtin NMS and I don't bother switching back. You need to pay attention to the details. 

This code can train on COCO very well, I've hit > 80 mAP fine-tuning on Pascal VOC, 1st, 2nd, and I think 5th place in a recent Kaggle challenge used this, and so did many other contestants. Someone used it for good results on a recent Waymo challenge. 

If you can isolate an actual bug, file one, but I don't have the capacity to help/tutor on custom datasets, etc.

**Comment by jinfagang** [comment]
@rwightman Thanks for your clarification, now I think it's some details I missed, your code should work. But I am a little confused by your so many branchs. Let me explain:

1. I am using master (haven't try your new branch yet), you said it's xywh, I didn't change anything in dataset.py, so I suppose if I using same annotation style as coco does, they should be identical? But in fact it's not.

2. which one is reasonablely and consistent both dataloader and nms and evaluation? xyxy? Will it work if switch to your new branch?
